iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -548,6 +552,32 @@ struct TileAndDistributeOnTensorsPattern
   }
 };
 
+/// The workload is computed based on the problem size. For a given operation,
+/// return the problem size.
+static Optional<SmallVector<Value, 4>> getProblemSize(PatternRewriter &rewriter,
+                                                      Operation *op) {
+  Location loc = op->getLoc();
+  auto getShapeOfShapedTypeVal = [&](Value v) -> SmallVector<Value, 4> {
+    return llvm::to_vector<4>(llvm::map_range(
+        llvm::seq<int64_t>(0, v.getType().cast<ShapedType>().getRank()),
+        [&](int64_t dim) -> Value {
+          return rewriter.create<DimOp>(loc, v, dim);
+        }));
+  };
```

Comment:
This is only used once. Can we don't make it a lambda?

Reply:
Actually I needed it in multiple places. But I deleted the other uses. Let me leave it for now. This gets cleaned up after the shape inference stuff lands.